### ISSUE TITLE
Reproducing validation results

### ISSUE THREAD

**Issue by JHLee0513** [rmokady/CLIP_prefix_caption#38]
Hello,

Thanks for the great work! I was interested in reproducing the transformer network with frozen GPT-2, and achieved slightly lower performance on COCO so far:

Metric | reported | reproduced
-- | -- | --
Bleu@4 | 33.53 | 31.0
METEOR | 27.45 | 27.1
CIDEr | 113.08 | 105.7
SPICE | 21.05 | 20.4

I was wondering if the provided code should be able to reproduce the validation scores or if I am missing something?

**Comment by rm-rf-me** [comment]
Hello, I am also going to try to train and test this model recently, but I am not very familiar with these evaluation indicators. Do you mainly rewrite the code when you are doing evaluation or refer to some external libraries?

**Comment by JHLee0513** [comment]
Hi @rm-rf-me , we used [COCO caption evaluation tool](https://github.com/LuoweiZhou/coco-caption/tree/de6f385503ac9a4305a1dcdc39c02312f9fa13fc) for our evaluation to get the above numbers

**Comment by rm-rf-me** [comment]
got it, thanks a lot !

**Comment by rmokady** [comment]
Hi @JHLee0513,
We have not validated that the code reproduce the exact numbers, but we did publish the model weights so you could check your evaluation script.
Please also note we have used Karpathy et el. script and the evaluation code of Oscar.

Is it possible that a bug or slightly different hyperparameters were inserted when publishing the code.
Sorry for not able to provide a solution to this issue.

**Comment by JHLee0513** [comment]
Hi @rmokady,
Thanks for the reply, I believe that's a good idea. It seems like there's only the pretrained model on resnet encoder, is there also one with ViT?

**Comment by rmokady** [comment]
Sorry, at this point we are not planning to release additional models

**Comment by JHLee0513** [comment]
Got it, thanks for the info!
